Assignments/RearrangingCars/RearrangeCars.java

```
@@ -0,0 +1,73 @@
+package RearrangingCars;
+
+import java.util.ArrayList;
+
+public class RearrangeCars {
+  public static void printCarMoves(int[] original, int[] target) {
+    ArrayList<Move> result = new ArrayList<>();
+    result = RearrangeCars(original, target);
+
+    for (Move m : result) {
+      m.printMove();
+    }
+  }
+  public static ArrayList<Move> RearrangeCars(int[] original, int[] target) {
+    if (original.length != target.length) {
+      return null;
```

I'd suggest throwing exceptions for bad input, rather than returning null.